packages/decoder/lib/ProviderAdapter.ts

```diff
@@ -72,74 +73,94 @@ const formatBlockSpecifier = (block: BlockSpecifier): string => {
   } else {
     throw new Error(
       "The block specified must be a number or one of the strings 'latest'," +
-      "'pending', or 'earliest'."
+        "'pending', or 'earliest'."
     );
   }
 };
 
+// EIP-1193 providers use `request()` instead of `send()`
+// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md
+const isEip1193Provider = (
+  provider: LegacyProvider | Eip1193Provider | EthersProvider
+): provider is Eip1193Provider => "request" in provider;
+
+// ethers' JsonRpcProvider's `send()` method takes separate method/params args
+// HACK detect this by looking for the ethers-specific `listAccounts` method
+// NOTE this provider returns `response.result` already unwrapped
+// https://docs.ethers.io/v5/api/providers/jsonrpc-provider/
+const isEthersProvider = (
+  provider: LegacyProvider | Eip1193Provider | EthersProvider
+): provider is EthersProvider => "listAccounts" in provider;
```

Comment:
Maybe?  I'd prefer to do things the right way from the start rather than fixing it later, given how small an effort it would be.  The change just consists of deleting this function and replacing it with an import, after all.  And this way if there *are* any problems from using ethers, we'll know about them immediately.

I mean if it were a big change then I'd be like yeah just merge this stopgap but it's a small change so I don't really see a reason?  Unless you actually think that importing ethers will cause significant enough LavaMoat problems that a stopgap is warranted.

Reply:
I don't think we need to support ethers providers as priority anymore (now that EIP-1559 providers work), so I'm good either way.

Reply:
Wait, "either way" being...?